util/deephash/deephash.go

```diff
@@ -326,17 +331,34 @@ func makeSliceHasher(t reflect.Type) typeHasherFunc {
 
 	var once sync.Once
 	var hashElem typeHasherFunc
+	var isRecursive bool
 	init := func() {
 		hashElem = getTypeInfo(t.Elem()).hasher()
+		isRecursive = typeIsRecursive(t)
 	}
 
 	return func(h *hasher, p pointer) {
 		pa := p.sliceArray()
+		if pa.isNil() {
```

Comment:
If the lack of nil-checking is deliberate, I'll put a comment here to the effect.
The `deepprint` predecessor never did nil-checks for slices,
but it feels weird and inconsistent that we do nil checks for maps, but not slices.

At the end of the day, the semantics here is more governed by how other packages use this,
and I don't have the best insight into that.